do not disconnect on receiving canceled request

etan-status opened the pull request:
LC cancels concurrent requests if one peer sent a correct response and
waiting for other peers is no longer useful. On the server side this
resulted in a descore (-500) and a likely disconnect. The behaviour is
changed to match `UnexpectedEOF`, `PotentiallyExpectedEOF` handling that
return an error response without disconnecting from the peer.

etan-status committed:
do not disconnect on receiving canceled request

LC cancels concurrent requests if one peer sent a correct response and
waiting for other peers is no longer useful. On the server side this
resulted in a descore (-500) and a likely disconnect. The behaviour is
changed to match `UnexpectedEOF`, `PotentiallyExpectedEOF` handling that
return an error response without disconnecting from the peer.

tersec commented on beacon_chain/networking/eth2_network.nim:
```diff
@@ -1024,8 +1024,11 @@ proc handleIncomingStream(network: Eth2Node,
           readChunkPayload(conn, peer, maxChunkSize(MsgRec), MsgRec), deadline):
-            returnInvalidRequest(
-              errorMsgLit "Request full data not sent in time")
+            # Request was canceled, e.g., because a different peer fulfilled it.
```
Is this timeout always a canceled request, rather than some other reason?

etan-status replied:
I guess the timeout could also be hit on slow connections, disconnections etc, i.e., network issues in general. But as we already have similar handling for other Eof behaviour further down that does not contain a descore, I guess it's fine to treat this similarly. I reworded the comment to mention a more broadly applicable "timeout" instead.

etan-status committed:
reword comment to more generic timeout

etan-status committed:
revert `requireAllFields` regression

#3864 introduced a regression by turning on `requireAllFields` globally
for JSON parsing. Certain endpoints such as `RestSyncInfo` have optional
fields that do not parse correctly without additional changes. This is
reverted for now to restore previous behaviour and unblock CI testing.

etan-status committed:
Merge branch 'dev/etan/ci-fixregressions' into dev/etan/nw-incompletedescore

etan-status committed:
Merge branch 'unstable' into dev/etan/nw-incompletedescore

etan-status committed:
Revert "revert `requireAllFields` regression"

This reverts commit 56f05252f820d765b76753c73ef14f4772230ef7.
